### PR TITLE
Implement set_scalar functions in the specialization constants C API

### PR DESCRIPTION
**Pull request by skyebird189**
This PR adds set_scalar functions for all equivalent get_scalar functions in the specialization constants part of the C API.

I've tested this and it seems to work fine.

### CODE REVIEW

File: spirv_cross_c.h [hunk 1]
```diff
@@ -1049,6 +1049,19 @@ SPVC_PUBLIC_API int spvc_constant_get_scalar_i8(spvc_constant constant, unsigned
...
+ */
+SPVC_PUBLIC_API void spvc_constant_set_scalar_fp16(spvc_constant constant, unsigned column, unsigned row, float value);
+SPVC_PUBLIC_API void spvc_constant_set_scalar_fp32(spvc_constant constant, unsigned column, unsigned row, float value);
+SPVC_PUBLIC_API void spvc_constant_set_scalar_fp64(spvc_constant constant, unsigned column, unsigned row, double value);
+SPVC_PUBLIC_API void spvc_constant_set_scalar_u32(spvc_constant constant, unsigned column, unsigned row, unsigned value);
+SPVC_PUBLIC_API void spvc_constant_set_scalar_i32(spvc_constant constant, unsigned column, unsigned row, int value);
+SPVC_PUBLIC_API void spvc_constant_set_scalar_u16(spvc_constant constant, unsigned column, unsigned row, unsigned value);
+SPVC_PUBLIC_API void spvc_constant_set_scalar_i16(spvc_constant constant, unsigned column, unsigned row, int value);
+SPVC_PUBLIC_API void spvc_constant_set_scalar_u8(spvc_constant constant, unsigned column, unsigned row, unsigned value);
+SPVC_PUBLIC_API void spvc_constant_set_scalar_i8(spvc_constant constant, unsigned column, unsigned row, int value);
```
Comment (HansKristian-Work) [inline spirv_cross_c.h:1063]: The types are wrong here:
8 -> unsigned and signed char
16 -> unsigned and signed short
FP16 -> unsigned short, it's supposed to be the raw fp16 payload, it gets decoded later in SPIRConstant::scalar_fp16().

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Mostly types being wrong, but otherwise ok to add this.

Also, remember to update SPVC_C_API_VERSION_MINOR and the corresponding value in CMakeLists.txt

**Comment by skyebird189** [comment]
I've made the requested changes, and hopefully fixed the types.

File: spirv_cross_c.cpp [hunk 2]
```diff
@@ -2535,6 +2535,51 @@ spvc_type_id spvc_constant_get_type(spvc_constant constant)
...
+}
+
+void spvc_constant_set_scalar_u16(spvc_constant constant, unsigned column, unsigned row, unsigned short value)
+{
+	constant->m.c[column].r[row].u32 = uint32_t(value);
+}
+
+void spvc_constant_set_scalar_i16(spvc_constant constant, unsigned column, unsigned row, signed short value)
+{
+	constant->m.c[column].r[row].u32 = uint32_t(value);
```
Comment (HansKristian-Work) [inline spirv_cross_c.cpp:2570]: need to cast to int32_t here, then uint32_t.

**Comment by skyebird189** [comment]
I've fixed the f16 type to use a u32, hopefully this is all good now!

**Review by HansKristian-Work — APPROVED** [review]
LGTM when CI passes.

**Pull request merged**